src/components/dev-hub/article-series.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import dlv from 'dlv';
+import Series from './series';
+
+const ArticleSeries = ({ allSeriesForArticle, slugTitleMapping, title }) => {
+    // Handle if this article is not in a series or no series are defined
+    if (!allSeriesForArticle) return null;
+    const getMappedSeries = seriesSlugs => {
+        if (!seriesSlugs || !seriesSlugs.length) return null;
+
+        let hasSeenActiveSlug = false;
+        const getSeriesArticlePosition = articleTitle => {
+            // Find if an article is the current one in the series, or is upcoming/past
+            if (hasSeenActiveSlug) {
+                return 'upcoming';
+            } else if (articleTitle === title) {
+                hasSeenActiveSlug = true;
+                return 'active';
+            }
+            return 'past';
+        };
+
+        const mappedSeries = seriesSlugs.map(slug => {
+            const title = dlv(slugTitleMapping, [slug, 0, 'value'], slug);
+            const position = getSeriesArticlePosition(title);
```

Comment:
Here is where I addressed the loop issue, I also pulled this component out because it was getting a bit long to leave in `article.js`.